lib/foreman_tasks/cleaner.rb

```diff
@@ -91,12 +95,28 @@ def tasks
     end
 
     def delete_tasks(chunk)
-      ForemanTasks::Task.where(:id => chunk.map(&:id)).delete_all
+      tasks = ForemanTasks::Task.where(:id => chunk.map(&:id))
+      tasks_to_csv(tasks, @backup_dir, 'foreman_tasks.csv') if @backup_dir
+      tasks.delete_all
+    end
+
+    def tasks_to_csv(dataset, backup_dir, file_name)
+      FileUtils.mkdir_p(backup_dir) unless File.directory?(backup_dir)
+      csv_file = File.join(backup_dir, file_name)
+      appending = File.exist?(csv_file)
+      columns = ForemanTasks::Task.attribute_names
+      File.open(csv_file, 'a') do |csv|
+        csv << columns.to_csv unless appending
+        dataset.each do |row|
+          csv << row.attributes.values.to_csv
+        end
+      end
+      dataset
     end
 
-    def delete_dynflow_plans(chunk)
+    def delete_dynflow_plans(chunk, backup_dir = ForemanTasks.dynflow.world.persistence.current_backup_dir)
```

Comment:
Any reason for passing the `backup_dir` explicitly instead of using `@backup_dir`?